compiler/rustc_typeck/src/check/method/confirm.rs

```diff
@@ -135,11 +136,13 @@ impl<'a, 'tcx> ConfirmContext<'a, 'tcx> {
         unadjusted_self_ty: Ty<'tcx>,
         pick: &probe::Pick<'tcx>,
     ) -> Ty<'tcx> {
+        debug!("adjust_self_ty unadjusted_self_ty={:#?}, pick={:#?}", unadjusted_self_ty, pick);
```

Comment:
I think it's possible to add some kind of "instrument" annotation now...

Reply:
You can add `#[instrument(skip(self))]` to the function attributes to get essentially the same dump as what you wrote here.

Reply:
I wanted to use the alternate style here, that's why I used `debug` instead of `instrument`.

Btw, I'm planning on removing all debug logs added in this PR, as mentioned in the description. Sorry the PR is a bit noisy with all these but I thought I'll need them again as the PR is not final yet.